File: app/models/qdm/tacoma/concept.rb
module Tacoma
  # Concept is an individual code within a ValueSet, including all of the associated metadata
  # such as code system, version, and display name
  class Concept
    include Mongoid::Document
    field :code, type: String
    field :code_system_oid, type: String
    field :code_system_name, type: String
    field :code_system_version, type: String
    field :display_name, type: String

    scope :by_code_system_name, ->(cs) { where(code_system_name: cs) }
    scope :by_code_system_oid, ->(cs) { where(code_system_oid: cs) }
  end
end

File: app/assets/javascripts/cqm/Concept.js
const mongoose = require('mongoose/browser');

const ConceptSchema = new mongoose.Schema({
  code: String,
  code_system_oid: String,
  code_system_name: String,
  code_system_version: String,
  display_name: String,
});

module.exports.ConceptSchema = ConceptSchema;
class Concept extends mongoose.Document {
  constructor(object) {
    super(object, ConceptSchema);
  }
}
module.exports.Concept = Concept;
